Flow.Launcher.Infrastructure/Http/Http.cs

```diff
@@ -6,85 +6,112 @@
                                                     | SecurityProtocolType.Tls11
                                                     | SecurityProtocolType.Tls12;
+
+            client = new HttpClient(socketsHttpHandler, false);
+            client.DefaultRequestHeaders.Add("User-Agent", UserAgent);
         }
 
-        public static HttpProxy Proxy { private get; set; }
-        public static IWebProxy WebProxy()
+        private static HttpProxy proxy;
+
+        public static HttpProxy Proxy
         {
-            if (Proxy != null && Proxy.Enabled && !string.IsNullOrEmpty(Proxy.Server))
+            private get { return proxy; }
+            set
             {
-                if (string.IsNullOrEmpty(Proxy.UserName) || string.IsNullOrEmpty(Proxy.Password))
-                {
-                    var webProxy = new WebProxy(Proxy.Server, Proxy.Port);
-                    return webProxy;
-                }
-                else
+                proxy = value;
+                proxy.PropertyChanged += UpdateProxy;
+            }
+        }
+
+        public static WebProxy WebProxy { get; } = new WebProxy();
+
+        /// <summary>
+        /// Update the Address of the Proxy to modify the client Proxy
+        /// </summary>
+        public static void UpdateProxy(ProxyProperty property)
+        {
+            (WebProxy.Address, WebProxy.Credentials) = property switch
+            {
+                ProxyProperty.Enabled => Proxy.Enabled switch
                 {
-                    var webProxy = new WebProxy(Proxy.Server, Proxy.Port)
+                    true => Proxy.UserName switch
                     {
-                        Credentials = new NetworkCredential(Proxy.UserName, Proxy.Password)
-                    };
-                    return webProxy;
-                }
+                        var userName when !string.IsNullOrEmpty(userName) =>
+                            (new Uri($"http://{Proxy.Server}:{Proxy.Port}"), null),
+                        _ => (new Uri($"http://{Proxy.Server}:{Proxy.Port}"),
+                            new NetworkCredential(Proxy.UserName, Proxy.Password))
+                    },
+                    false => (null, null)
+                },
+                ProxyProperty.Server => (new Uri($"http://{Proxy.Server}:{Proxy.Port}"), WebProxy.Credentials),
+                ProxyProperty.Port => (new Uri($"http://{Proxy.Server}:{Proxy.Port}"), WebProxy.Credentials),
+                ProxyProperty.UserName => (WebProxy.Address, new NetworkCredential(Proxy.UserName, Proxy.Password)),
+                ProxyProperty.Password => (WebProxy.Address, new NetworkCredential(Proxy.UserName, Proxy.Password)),
+                _ => throw new ArgumentOutOfRangeException()
+            };
+        }
+
+        public static async Task Download([NotNull] string url, [NotNull] string filePath)
+        {
+            using var response = await client.GetAsync(url);
```

Comment:
client.GetAsync crashes the entire app for me when i try to install plugin. Cant see any reason why though, no exception is thrown

Reply:
Let me take a check.

Reply:
Oh that's because before the Http.Download is not async, so no await is added there.